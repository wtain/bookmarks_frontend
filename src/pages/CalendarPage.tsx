
import React, { useEffect, useState } from "react"
// import { Calendar, Views, momentLocalizer } from 'react-big-calendar'
import moment from 'moment'
import Calendar from 'react-calendar'
import 'react-calendar/dist/Calendar.css';
import cl from './CalendarPage.module.css'

const CalendarPage = () => {

    // const localizer = momentLocalizer(moment) // or globalizeLocalizer

    return (
        <div className={cl.main_div}>
            {/* <Calendar localizer={localizer} /> */}
            <Calendar onChange={(date: Date) => {

            }} />
        </div>
    )
}

export default CalendarPage;