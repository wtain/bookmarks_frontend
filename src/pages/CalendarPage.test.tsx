import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import CalendarPage from './CalendarPage';

it('CalendarPage renders correctly', () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <CalendarPage />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});