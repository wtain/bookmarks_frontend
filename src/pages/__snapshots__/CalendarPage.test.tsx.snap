// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarPage renders correctly 1`] = `
<div
  className="main_div"
>
  <div
    className="react-calendar"
  >
    <div
      className="react-calendar__navigation"
    >
      <button
        aria-label=""
        className="react-calendar__navigation__arrow react-calendar__navigation__prev2-button"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        «
      </button>
      <button
        aria-label=""
        className="react-calendar__navigation__arrow react-calendar__navigation__prev-button"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        ‹
      </button>
      <button
        aria-label=""
        className="react-calendar__navigation__label"
        disabled={false}
        onClick={[Function]}
        style={
          Object {
            "flexGrow": 1,
          }
        }
        type="button"
      >
        <span
          className="react-calendar__navigation__label__labelText react-calendar__navigation__label__labelText--from"
        >
          September 2022
        </span>
      </button>
      <button
        aria-label=""
        className="react-calendar__navigation__arrow react-calendar__navigation__next-button"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        ›
      </button>
      <button
        aria-label=""
        className="react-calendar__navigation__arrow react-calendar__navigation__next2-button"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        »
      </button>
    </div>
    <div
      className="react-calendar__viewContainer"
    >
      <div
        className="react-calendar__month-view"
      >
        <div
          style={
            Object {
              "alignItems": "flex-end",
              "display": "flex",
            }
          }
        >
          <div
            style={
              Object {
                "flexGrow": 1,
                "width": "100%",
              }
            }
          >
            <div
              className="react-calendar__month-view__weekdays"
              onFocus={null}
              onMouseOver={null}
              style={
                Object {
                  "display": "flex",
                  "flexDirection": undefined,
                  "flexWrap": "nowrap",
                }
              }
            >
              <div
                className="react-calendar__month-view__weekdays__weekday"
                style={
                  Object {
                    "flexBasis": "14.285714285714286%",
                    "flexGrow": 0,
                    "flexShrink": 0,
                    "marginLeft": null,
                    "overflow": "hidden",
                  }
                }
              >
                <abbr
                  aria-label="Monday"
                  title="Monday"
                >
                  Mon
                </abbr>
              </div>
              <div
                className="react-calendar__month-view__weekdays__weekday"
                style={
                  Object {
                    "flexBasis": "14.285714285714286%",
                    "flexGrow": 0,
                    "flexShrink": 0,
                    "marginLeft": null,
                    "overflow": "hidden",
                  }
                }
              >
                <abbr
                  aria-label="Tuesday"
                  title="Tuesday"
                >
                  Tue
                </abbr>
              </div>
              <div
                className="react-calendar__month-view__weekdays__weekday"
                style={
                  Object {
                    "flexBasis": "14.285714285714286%",
                    "flexGrow": 0,
                    "flexShrink": 0,
                    "marginLeft": null,
                    "overflow": "hidden",
                  }
                }
              >
                <abbr
                  aria-label="Wednesday"
                  title="Wednesday"
                >
                  Wed
                </abbr>
              </div>
              <div
                className="react-calendar__month-view__weekdays__weekday"
                style={
                  Object {
                    "flexBasis": "14.285714285714286%",
                    "flexGrow": 0,
                    "flexShrink": 0,
                    "marginLeft": null,
                    "overflow": "hidden",
                  }
                }
              >
                <abbr
                  aria-label="Thursday"
                  title="Thursday"
                >
                  Thu
                </abbr>
              </div>
              <div
                className="react-calendar__month-view__weekdays__weekday"
                style={
                  Object {
                    "flexBasis": "14.285714285714286%",
                    "flexGrow": 0,
                    "flexShrink": 0,
                    "marginLeft": null,
                    "overflow": "hidden",
                  }
                }
              >
                <abbr
                  aria-label="Friday"
                  title="Friday"
                >
                  Fri
                </abbr>
              </div>
              <div
                className="react-calendar__month-view__weekdays__weekday react-calendar__month-view__weekdays__weekday--current react-calendar__month-view__weekdays__weekday--weekend"
                style={
                  Object {
                    "flexBasis": "14.285714285714286%",
                    "flexGrow": 0,
                    "flexShrink": 0,
                    "marginLeft": null,
                    "overflow": "hidden",
                  }
                }
              >
                <abbr
                  aria-label="Saturday"
                  title="Saturday"
                >
                  Sat
                </abbr>
              </div>
              <div
                className="react-calendar__month-view__weekdays__weekday react-calendar__month-view__weekdays__weekday--weekend"
                style={
                  Object {
                    "flexBasis": "14.285714285714286%",
                    "flexGrow": 0,
                    "flexShrink": 0,
                    "marginLeft": null,
                    "overflow": "hidden",
                  }
                }
              >
                <abbr
                  aria-label="Sunday"
                  title="Sunday"
                >
                  Sun
                </abbr>
              </div>
            </div>
            <div
              className="react-calendar__month-view__days"
              style={
                Object {
                  "display": "flex",
                  "flexDirection": undefined,
                  "flexWrap": "wrap",
                }
              }
            >
              <button
                className="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--neighboringMonth"
                onClick={[Function]}
                style={
                  Object {
                    "flexBasis": "14.285714285714286%",
                    "flexGrow": 0,
                    "flexShrink": 0,
                    "marginLeft": null,
                    "overflow": "hidden",
                  }
                }
                type="button"
              >
                <abbr
                  aria-label="August 29, 2022"
                >
                  29
                </abbr>
              </button>
              <button
                className="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--neighboringMonth"
                onClick={[Function]}
                style={
                  Object {
                    "flexBasis": "14.285714285714286%",
                    "flexGrow": 0,
                    "flexShrink": 0,
                    "marginLeft": null,
                    "overflow": "hidden",
                  }
                }
                type="button"
              >
                <abbr
                  aria-label="August 30, 2022"
                >
                  30
                </abbr>
              </button>
              <button
                className="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--neighboringMonth"
                onClick={[Function]}
                style={
                  Object {
                    "flexBasis": "14.285714285714286%",
                    "flexGrow": 0,
                    "flexShrink": 0,
                    "marginLeft": null,
                    "overflow": "hidden",
                  }
                }
                type="button"
              >
                <abbr
                  aria-label="August 31, 2022"
                >
                  31
                </abbr>
              </button>
              <button
                className="react-calendar__tile react-calendar__month-view__days__day"
                onClick={[Function]}
                style={
                  Object {
                    "flexBasis": "14.285714285714286%",
                    "flexGrow": 0,
                    "flexShrink": 0,
                    "marginLeft": null,
                    "overflow": "hidden",
                  }
                }
                type="button"
              >
                <abbr
                  aria-label="September 1, 2022"
                >
                  1
                </abbr>
              </button>
              <button
                className="react-calendar__tile react-calendar__month-view__days__day"
                onClick={[Function]}
                style={
                  Object {
                    "flexBasis": "14.285714285714286%",
                    "flexGrow": 0,
                    "flexShrink": 0,
                    "marginLeft": null,
                    "overflow": "hidden",
                  }
                }
                type="button"
              >
                <abbr
                  aria-label="September 2, 2022"
                >
                  2
                </abbr>
              </button>
              <button
                className="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                onClick={[Function]}
                style={
                  Object {
                    "flexBasis": "14.285714285714286%",
                    "flexGrow": 0,
                    "flexShrink": 0,
                    "marginLeft": null,
                    "overflow": "hidden",
                  }
                }
                type="button"
              >
                <abbr
                  aria-label="September 3, 2022"
                >
                  3
                </abbr>
              </button>
              <button
                className="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                onClick={[Function]}
                style={
                  Object {
                    "flexBasis": "14.285714285714286%",
                    "flexGrow": 0,
                    "flexShrink": 0,
                    "marginLeft": null,
                    "overflow": "hidden",
                  }
                }
                type="button"
              >
                <abbr
                  aria-label="September 4, 2022"
                >
                  4
                </abbr>
              </button>
              <button
                className="react-calendar__tile react-calendar__month-view__days__day"
                onClick={[Function]}
                style={
                  Object {
                    "flexBasis": "14.285714285714286%",
                    "flexGrow": 0,
                    "flexShrink": 0,
                    "marginLeft": null,
                    "overflow": "hidden",
                  }
                }
                type="button"
              >
                <abbr
                  aria-label="September 5, 2022"
                >
                  5
                </abbr>
              </button>
              <button
                className="react-calendar__tile react-calendar__month-view__days__day"
                onClick={[Function]}
                style={
                  Object {
                    "flexBasis": "14.285714285714286%",
                    "flexGrow": 0,
                    "flexShrink": 0,
                    "marginLeft": null,
                    "overflow": "hidden",
                  }
                }
                type="button"
              >
                <abbr
                  aria-label="September 6, 2022"
                >
                  6
                </abbr>
              </button>
              <button
                className="react-calendar__tile react-calendar__month-view__days__day"
                onClick={[Function]}
                style={
                  Object {
                    "flexBasis": "14.285714285714286%",
                    "flexGrow": 0,
                    "flexShrink": 0,
                    "marginLeft": null,
                    "overflow": "hidden",
                  }
                }
                type="button"
              >
                <abbr
                  aria-label="September 7, 2022"
                >
                  7
                </abbr>
              </button>
              <button
                className="react-calendar__tile react-calendar__month-view__days__day"
                onClick={[Function]}
                style={
                  Object {
                    "flexBasis": "14.285714285714286%",
                    "flexGrow": 0,
                    "flexShrink": 0,
                    "marginLeft": null,
                    "overflow": "hidden",
                  }
                }
                type="button"
              >
                <abbr
                  aria-label="September 8, 2022"
                >
                  8
                </abbr>
              </button>
              <button
                className="react-calendar__tile react-calendar__month-view__days__day"
                onClick={[Function]}
                style={
                  Object {
                    "flexBasis": "14.285714285714286%",
                    "flexGrow": 0,
                    "flexShrink": 0,
                    "marginLeft": null,
                    "overflow": "hidden",
                  }
                }
                type="button"
              >
                <abbr
                  aria-label="September 9, 2022"
                >
                  9
                </abbr>
              </button>
              <button
                className="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                onClick={[Function]}
                style={
                  Object {
                    "flexBasis": "14.285714285714286%",
                    "flexGrow": 0,
                    "flexShrink": 0,
                    "marginLeft": null,
                    "overflow": "hidden",
                  }
                }
                type="button"
              >
                <abbr
                  aria-label="September 10, 2022"
                >
                  10
                </abbr>
              </button>
              <button
                className="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                onClick={[Function]}
                style={
                  Object {
                    "flexBasis": "14.285714285714286%",
                    "flexGrow": 0,
                    "flexShrink": 0,
                    "marginLeft": null,
                    "overflow": "hidden",
                  }
                }
                type="button"
              >
                <abbr
                  aria-label="September 11, 2022"
                >
                  11
                </abbr>
              </button>
              <button
                className="react-calendar__tile react-calendar__month-view__days__day"
                onClick={[Function]}
                style={
                  Object {
                    "flexBasis": "14.285714285714286%",
                    "flexGrow": 0,
                    "flexShrink": 0,
                    "marginLeft": null,
                    "overflow": "hidden",
                  }
                }
                type="button"
              >
                <abbr
                  aria-label="September 12, 2022"
                >
                  12
                </abbr>
              </button>
              <button
                className="react-calendar__tile react-calendar__month-view__days__day"
                onClick={[Function]}
                style={
                  Object {
                    "flexBasis": "14.285714285714286%",
                    "flexGrow": 0,
                    "flexShrink": 0,
                    "marginLeft": null,
                    "overflow": "hidden",
                  }
                }
                type="button"
              >
                <abbr
                  aria-label="September 13, 2022"
                >
                  13
                </abbr>
              </button>
              <button
                className="react-calendar__tile react-calendar__month-view__days__day"
                onClick={[Function]}
                style={
                  Object {
                    "flexBasis": "14.285714285714286%",
                    "flexGrow": 0,
                    "flexShrink": 0,
                    "marginLeft": null,
                    "overflow": "hidden",
                  }
                }
                type="button"
              >
                <abbr
                  aria-label="September 14, 2022"
                >
                  14
                </abbr>
              </button>
              <button
                className="react-calendar__tile react-calendar__month-view__days__day"
                onClick={[Function]}
                style={
                  Object {
                    "flexBasis": "14.285714285714286%",
                    "flexGrow": 0,
                    "flexShrink": 0,
                    "marginLeft": null,
                    "overflow": "hidden",
                  }
                }
                type="button"
              >
                <abbr
                  aria-label="September 15, 2022"
                >
                  15
                </abbr>
              </button>
              <button
                className="react-calendar__tile react-calendar__month-view__days__day"
                onClick={[Function]}
                style={
                  Object {
                    "flexBasis": "14.285714285714286%",
                    "flexGrow": 0,
                    "flexShrink": 0,
                    "marginLeft": null,
                    "overflow": "hidden",
                  }
                }
                type="button"
              >
                <abbr
                  aria-label="September 16, 2022"
                >
                  16
                </abbr>
              </button>
              <button
                className="react-calendar__tile react-calendar__tile--now react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                onClick={[Function]}
                style={
                  Object {
                    "flexBasis": "14.285714285714286%",
                    "flexGrow": 0,
                    "flexShrink": 0,
                    "marginLeft": null,
                    "overflow": "hidden",
                  }
                }
                type="button"
              >
                <abbr
                  aria-label="September 17, 2022"
                >
                  17
                </abbr>
              </button>
              <button
                className="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                onClick={[Function]}
                style={
                  Object {
                    "flexBasis": "14.285714285714286%",
                    "flexGrow": 0,
                    "flexShrink": 0,
                    "marginLeft": null,
                    "overflow": "hidden",
                  }
                }
                type="button"
              >
                <abbr
                  aria-label="September 18, 2022"
                >
                  18
                </abbr>
              </button>
              <button
                className="react-calendar__tile react-calendar__month-view__days__day"
                onClick={[Function]}
                style={
                  Object {
                    "flexBasis": "14.285714285714286%",
                    "flexGrow": 0,
                    "flexShrink": 0,
                    "marginLeft": null,
                    "overflow": "hidden",
                  }
                }
                type="button"
              >
                <abbr
                  aria-label="September 19, 2022"
                >
                  19
                </abbr>
              </button>
              <button
                className="react-calendar__tile react-calendar__month-view__days__day"
                onClick={[Function]}
                style={
                  Object {
                    "flexBasis": "14.285714285714286%",
                    "flexGrow": 0,
                    "flexShrink": 0,
                    "marginLeft": null,
                    "overflow": "hidden",
                  }
                }
                type="button"
              >
                <abbr
                  aria-label="September 20, 2022"
                >
                  20
                </abbr>
              </button>
              <button
                className="react-calendar__tile react-calendar__month-view__days__day"
                onClick={[Function]}
                style={
                  Object {
                    "flexBasis": "14.285714285714286%",
                    "flexGrow": 0,
                    "flexShrink": 0,
                    "marginLeft": null,
                    "overflow": "hidden",
                  }
                }
                type="button"
              >
                <abbr
                  aria-label="September 21, 2022"
                >
                  21
                </abbr>
              </button>
              <button
                className="react-calendar__tile react-calendar__month-view__days__day"
                onClick={[Function]}
                style={
                  Object {
                    "flexBasis": "14.285714285714286%",
                    "flexGrow": 0,
                    "flexShrink": 0,
                    "marginLeft": null,
                    "overflow": "hidden",
                  }
                }
                type="button"
              >
                <abbr
                  aria-label="September 22, 2022"
                >
                  22
                </abbr>
              </button>
              <button
                className="react-calendar__tile react-calendar__month-view__days__day"
                onClick={[Function]}
                style={
                  Object {
                    "flexBasis": "14.285714285714286%",
                    "flexGrow": 0,
                    "flexShrink": 0,
                    "marginLeft": null,
                    "overflow": "hidden",
                  }
                }
                type="button"
              >
                <abbr
                  aria-label="September 23, 2022"
                >
                  23
                </abbr>
              </button>
              <button
                className="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                onClick={[Function]}
                style={
                  Object {
                    "flexBasis": "14.285714285714286%",
                    "flexGrow": 0,
                    "flexShrink": 0,
                    "marginLeft": null,
                    "overflow": "hidden",
                  }
                }
                type="button"
              >
                <abbr
                  aria-label="September 24, 2022"
                >
                  24
                </abbr>
              </button>
              <button
                className="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                onClick={[Function]}
                style={
                  Object {
                    "flexBasis": "14.285714285714286%",
                    "flexGrow": 0,
                    "flexShrink": 0,
                    "marginLeft": null,
                    "overflow": "hidden",
                  }
                }
                type="button"
              >
                <abbr
                  aria-label="September 25, 2022"
                >
                  25
                </abbr>
              </button>
              <button
                className="react-calendar__tile react-calendar__month-view__days__day"
                onClick={[Function]}
                style={
                  Object {
                    "flexBasis": "14.285714285714286%",
                    "flexGrow": 0,
                    "flexShrink": 0,
                    "marginLeft": null,
                    "overflow": "hidden",
                  }
                }
                type="button"
              >
                <abbr
                  aria-label="September 26, 2022"
                >
                  26
                </abbr>
              </button>
              <button
                className="react-calendar__tile react-calendar__month-view__days__day"
                onClick={[Function]}
                style={
                  Object {
                    "flexBasis": "14.285714285714286%",
                    "flexGrow": 0,
                    "flexShrink": 0,
                    "marginLeft": null,
                    "overflow": "hidden",
                  }
                }
                type="button"
              >
                <abbr
                  aria-label="September 27, 2022"
                >
                  27
                </abbr>
              </button>
              <button
                className="react-calendar__tile react-calendar__month-view__days__day"
                onClick={[Function]}
                style={
                  Object {
                    "flexBasis": "14.285714285714286%",
                    "flexGrow": 0,
                    "flexShrink": 0,
                    "marginLeft": null,
                    "overflow": "hidden",
                  }
                }
                type="button"
              >
                <abbr
                  aria-label="September 28, 2022"
                >
                  28
                </abbr>
              </button>
              <button
                className="react-calendar__tile react-calendar__month-view__days__day"
                onClick={[Function]}
                style={
                  Object {
                    "flexBasis": "14.285714285714286%",
                    "flexGrow": 0,
                    "flexShrink": 0,
                    "marginLeft": null,
                    "overflow": "hidden",
                  }
                }
                type="button"
              >
                <abbr
                  aria-label="September 29, 2022"
                >
                  29
                </abbr>
              </button>
              <button
                className="react-calendar__tile react-calendar__month-view__days__day"
                onClick={[Function]}
                style={
                  Object {
                    "flexBasis": "14.285714285714286%",
                    "flexGrow": 0,
                    "flexShrink": 0,
                    "marginLeft": null,
                    "overflow": "hidden",
                  }
                }
                type="button"
              >
                <abbr
                  aria-label="September 30, 2022"
                >
                  30
                </abbr>
              </button>
              <button
                className="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend react-calendar__month-view__days__day--neighboringMonth"
                onClick={[Function]}
                style={
                  Object {
                    "flexBasis": "14.285714285714286%",
                    "flexGrow": 0,
                    "flexShrink": 0,
                    "marginLeft": null,
                    "overflow": "hidden",
                  }
                }
                type="button"
              >
                <abbr
                  aria-label="October 1, 2022"
                >
                  1
                </abbr>
              </button>
              <button
                className="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend react-calendar__month-view__days__day--neighboringMonth"
                onClick={[Function]}
                style={
                  Object {
                    "flexBasis": "14.285714285714286%",
                    "flexGrow": 0,
                    "flexShrink": 0,
                    "marginLeft": null,
                    "overflow": "hidden",
                  }
                }
                type="button"
              >
                <abbr
                  aria-label="October 2, 2022"
                >
                  2
                </abbr>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
